cdap-common/src/main/java/co/cask/cdap/internal/io/ReflectionPutWriter.java

```diff
@@ -116,31 +116,27 @@ protected void writeBytes(Put put, byte[] val) throws IOException {
 
   @Override
   protected void writeEnum(Put put, String val, Schema schema) throws IOException {
-    int idx = schema.getEnumIndex(val);
-    if (idx < 0) {
-      throw new IOException("Invalid enum value " + val);
-    }
-    put.add(nextField(), idx);
+    put.add(nextField(), val);
```

Comment:
that and we when we are deserializing we don't have the type token to create the actual enum.  Actually, I had not intended to support this because I didn't want to set a precendent here without thinking about it more, but forgot to replace this with an unsupported exception.